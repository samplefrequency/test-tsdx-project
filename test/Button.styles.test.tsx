import React from 'react';
import StyledButton from '../src/Button/Button.styles';
import 'jest-styled-components';
import { render, screen } from '@testing-library/react';

describe('Default styled button', () => {
  it('renders without error', () => {
    render(<StyledButton>test</StyledButton>);
    expect(screen.getByRole('button', { name: /test/i })).toBeInTheDocument();
  });

  it('renders default styles', () => {
    render(<StyledButton>test</StyledButton>);
    const button = screen.getByRole('button', { name: /test/i });
    expect(button).toHaveStyleRule(
      'background-color',
      'var(--color-white)',
      {}
    );
    expect(button).toHaveStyleRule(
      'border',
      '1px solid var(--color-brand-primary)'
    );

    expect(button).toHaveStyleRule(
      'background-color',
      'var(--color-brand-secondary-hover)',
      { modifier: ':hover' }
    );
  });

  it('matches snapshot', () => {
    const { container } = render(<StyledButton>test</StyledButton>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
