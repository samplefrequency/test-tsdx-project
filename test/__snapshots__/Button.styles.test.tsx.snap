// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default styled button matches snapshot 1`] = `
.c0 {
  background-color: var(--color-white);
  border: 1px solid var(--color-brand-primary);
  color: var(--color-brand-primary);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.375rem;
  box-shadow: 0 1px 2px rgba(0,0,0,0.05);
  margin: 0;
  outline: none;
  padding: 0.75rem;
}

.c0:hover {
  background-color: var(--color-brand-secondary-hover);
  cursor: pointer;
}

.c0:active {
  background-color: var(--color-brand-secondary-pressed);
}

<button
  class="c0"
>
  test
</button>
`;
